source/code/providers/Container_DaemonEvent_Class_Provider.cpp

```diff
@@ -79,7 +86,13 @@ class EventQuery
 	///
 	static void SetPreviousTime(int t)
 	{
-		FILE* file = fopen(LASTQUERYTIMEFILE, "w");
+		char* lastQueryFile = LASTQUERYTIMEFILE;
+		const char *cTestRun = getenv("CONTAINER_TESTRUN_ACTIVE");
+		if (cTestRun != NULL)
+		{
+			lastQueryFile = TEST_LASTQUERYTIMEFILE;
+		}
+		FILE* file = fopen(lastQueryFile, "w");

```

Comment:
Code duplication. Consider creating a function that returns a FILE* (and, I guess, a lastQueryFile pointer by parameter), and then calling that here and above. Less work if we need to change something.